### PR TITLE
Support aggregate functions in Eval expressions

### PR DESCRIPTION
**Pull request by LantaoJin**
### Description
Aggregate functions could work in select clause even there is no group by as long as all items in `select` are aggregate functions.
Here are some examples:
```sql
select max(a) from t
select max(a), min(a), count(a) from t
```
But the following queries should throw exceptions
```sql
select a, max(a) from t
select a, max(a), min(a), count(a) from t
select *, max(a), min(a), count(a) from t
```
They could work with a `group by`
```sql
select a, max(a) from t group by b
select a, max(a), min(a), count(a) from t group by b
select *, max(a), min(a), count(a) from t group by b
```

Similar, aggregate functions in PPL could work in `eval` command, because an eval expression equals to add a projection to existing project list.
Here are some examples:
```sql
source=t | eval m = max(a) | fields m
source=t | eval m = max(a), n = count(a) | fields m, n
source=t | eval m = max(a) | n = count(a) | fields m, n
```
But the following PPL queries should throw exceptions
```sql
source=t | eval m = max(a) -- equals to all fields plus m
source=t | eval m = max(a), n = count(a) | fields a, m, n
source=t | eval m = max(a) | n = count(a) | fields a, m, n
```

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/755

### Check List
~- [ ] Updated documentation (ppl-spark-integration/README.md)~
- [x] Implemented unit tests
- [x] Implemented tests for combination with other commands
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanEvalTranslatorTestSuite.scala [hunk 1]
```diff
@@ -177,27 +267,30 @@ class PPLLogicalPlanEvalTranslatorTestSuite
     comparePlans(expectedPlan, logPlan, checkAnalysis = false)
   }
 
-  // Todo fields-excluded command not supported
-  ignore("test eval expressions with fields-excluded command") {
+  test("test eval expressions with fields-excluded command") {
```
Comment (LantaoJin) [inline ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanEvalTranslatorTestSuite.scala:270]: This change is not related. Just enable the test since fields exclude list has been supported.

Comment (LantaoJin) [reply ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanEvalTranslatorTestSuite.scala:270]: But I can revert this if we want backport to 0.5-nexus since this changes import `DataFrameDropColumns` which unsupported in 0.5-nexus. Any thoughts? @YANG-DB

**Comment by YANG-DB** [comment]
@LantaoJin this is great !
can you please add the following paragraph to the [stats command document ](https://github.com/opensearch-project/opensearch-spark/blob/main/docs/ppl-lang/ppl-stats-command.md)?

```
Description
Aggregate functions could work in select clause even there is no group by as long as all items in select are aggregate functions.
Here are some examples:

select max(a) from t
select max(a), min(a), count(a) from t
But the following queries should throw exceptions

select a, max(a) from t
select a, max(a), min(a), count(a) from t
select *, max(a), min(a), count(a) from t
They could work with a group by

select a, max(a) from t group by b
select a, max(a), min(a), count(a) from t group by b
select *, max(a), min(a), count(a) from t group by b
Similar, aggregate functions in PPL could work in eval command, because an eval expression equals to add a projection to existing project list.
Here are some examples:

source=t | eval m = max(a) | fields m
source=t | eval m = max(a), n = count(a) | fields m, n
source=t | eval m = max(a) | n = count(a) | fields m, n
But the following PPL queries should throw exceptions

source=t | eval m = max(a) -- equals to all fields plus m
source=t | eval m = max(a), n = count(a) | fields a, m, n
source=t | eval m = max(a) | n = count(a) | fields a, m, n
```

**Review by YANG-DB — APPROVED** [review]

**Comment by LantaoJin** [comment]
I am going to close this due to https://github.com/opensearch-project/opensearch-spark/issues/755#issuecomment-2404539545
